pandas/core/sparse/series.py

```diff
@@ -146,10 +146,10 @@ def __init__(self, data=None, index=None, sparse_index=None, kind='block',
                 data = data._data
 
             elif isinstance(data, (Series, dict)):
+                data = Series(data, index=index)
                 if index is None:
```

Comment:
I ran the full test suite with this change and everything is working. I also manually ran a bunch of SparseSeries with different inputs (Series, dicts, with/without indexes) and it does work as intended.